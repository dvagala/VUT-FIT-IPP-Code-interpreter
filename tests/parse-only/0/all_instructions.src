.IPPcode19

MOVE GF@var GF@var
MOVE GF@var string@aa
MOVE GF@var nil@nil
MOVE GF@var int@8 

CREATEFRAME

POPFRAME

DEFVAR GF@var

CALL label
CALL MOVE
CALL string
RETURN
PUSHS  GF@var
PUSHS string@\010
POPS GF@var

ADD GF@var GF@var GF@var
ADD GF@var string@aa GF@var
ADD GF@var GF@var nil@nil
ADD GF@var bool@true int@8

SUB GF@var GF@var GF@var
SUB GF@var string@aa GF@var
SUB GF@var GF@var nil@nil
SUB GF@var bool@true int@8

MUL GF@var GF@var GF@var
MUL GF@var string@aa GF@var
MUL GF@var GF@var nil@nil
MUL GF@var bool@true int@8

IDIV GF@var GF@var GF@var
IDIV GF@var string@aa GF@var
IDIV GF@var GF@var nil@nil
IDIV GF@var bool@true int@8

LT GF@var GF@var GF@var
GT GF@var string@aa GF@var
EQ GF@var GF@var nil@nil
LT GF@var bool@true int@8

AND GF@var GF@var GF@var
OR GF@var string@aa GF@var
NOT GF@var GF@var
NOT GF@var bool@true

INT2CHAR GF@var GF@var
INT2CHAR GF@var string@aa

STRI2INT GF@var GF@var GF@var
STRI2INT GF@var string@aa GF@var
STRI2INT GF@var GF@var nil@nil
STRI2INT GF@var bool@true int@8

READ GF@var string
READ GF@var int
READ GF@var bool

WRITE GF@var
WRITE string@aa
WRITE int@8
WRITE bool@true
WRITE nil@nil

CONCAT GF@var GF@var GF@var
CONCAT GF@var string@aa GF@var
CONCAT GF@var GF@var nil@nil
CONCAT GF@var bool@true int@8

STRLEN GF@var GF@var
STRLEN GF@var string@aa
STRLEN GF@var nil@nil
STRLEN GF@var int@8 

GETCHAR GF@var GF@var GF@var
GETCHAR GF@var string@aa GF@var
GETCHAR GF@var GF@var nil@nil
GETCHAR GF@var bool@true int@8

SETCHAR GF@var GF@var GF@var
SETCHAR GF@var string@aa GF@var
SETCHAR GF@var GF@var nil@nil
SETCHAR GF@var bool@true int@8

TYPE GF@var GF@var
TYPE GF@var string@aa
TYPE GF@var nil@nil
TYPE GF@var int@8 

LABEL label
JUMP label

JUMPIFEQ label GF@var GF@var
JUMPIFEQ label string@aa GF@var
JUMPIFEQ label GF@var nil@nil
JUMPIFEQ label bool@true int@8

JUMPIFNEQ label GF@var GF@var
JUMPIFNEQ label string@aa GF@var
JUMPIFNEQ label GF@var nil@nil
JUMPIFNEQ label bool@true int@8

EXIT GF@var
EXIT string@aa
EXIT int@8
EXIT bool@true
EXIT nil@nil

DPRINT GF@var
DPRINT string@aa
DPRINT int@8
DPRINT bool@true
DPRINT nil@nil

BREAK
