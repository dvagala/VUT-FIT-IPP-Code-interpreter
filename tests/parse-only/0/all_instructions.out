<?xml version="1.0" encoding="UTF-8"?>
<program language="IPPcode19">
   <instruction opcode="MOVE" order="1">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
   </instruction>
   <instruction opcode="MOVE" order="2">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
   </instruction>
   <instruction opcode="MOVE" order="3">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="nil">nil</arg2>
   </instruction>
   <instruction opcode="MOVE" order="4">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="int">8</arg2>
   </instruction>
   <instruction opcode="CREATEFRAME" order="5"/>
   <instruction opcode="POPFRAME" order="6"/>
   <instruction opcode="DEFVAR" order="7">
      <arg1 type="var">GF@var</arg1>
   </instruction>
   <instruction opcode="CALL" order="8">
      <arg1 type="label">label</arg1>
   </instruction>
   <instruction opcode="CALL" order="9">
      <arg1 type="label">MOVE</arg1>
   </instruction>
   <instruction opcode="CALL" order="10">
      <arg1 type="label">string</arg1>
   </instruction>
   <instruction opcode="RETURN" order="11"/>
   <instruction opcode="PUSHS" order="12">
      <arg1 type="var">GF@var</arg1>
   </instruction>
   <instruction opcode="PUSHS" order="13">
      <arg1 type="string">\010</arg1>
   </instruction>
   <instruction opcode="POPS" order="14">
      <arg1 type="var">GF@var</arg1>
   </instruction>
   <instruction opcode="ADD" order="15">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="ADD" order="16">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="ADD" order="17">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="nil">nil</arg3>
   </instruction>
   <instruction opcode="ADD" order="18">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="bool">true</arg2>
      <arg3 type="int">8</arg3>
   </instruction>
   <instruction opcode="SUB" order="19">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="SUB" order="20">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="SUB" order="21">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="nil">nil</arg3>
   </instruction>
   <instruction opcode="SUB" order="22">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="bool">true</arg2>
      <arg3 type="int">8</arg3>
   </instruction>
   <instruction opcode="MUL" order="23">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="MUL" order="24">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="MUL" order="25">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="nil">nil</arg3>
   </instruction>
   <instruction opcode="MUL" order="26">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="bool">true</arg2>
      <arg3 type="int">8</arg3>
   </instruction>
   <instruction opcode="IDIV" order="27">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="IDIV" order="28">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="IDIV" order="29">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="nil">nil</arg3>
   </instruction>
   <instruction opcode="IDIV" order="30">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="bool">true</arg2>
      <arg3 type="int">8</arg3>
   </instruction>
   <instruction opcode="LT" order="31">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="GT" order="32">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="EQ" order="33">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="nil">nil</arg3>
   </instruction>
   <instruction opcode="LT" order="34">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="bool">true</arg2>
      <arg3 type="int">8</arg3>
   </instruction>
   <instruction opcode="AND" order="35">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="OR" order="36">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="NOT" order="37">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
   </instruction>
   <instruction opcode="NOT" order="38">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="bool">true</arg2>
   </instruction>
   <instruction opcode="INT2CHAR" order="39">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
   </instruction>
   <instruction opcode="INT2CHAR" order="40">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
   </instruction>
   <instruction opcode="STRI2INT" order="41">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="STRI2INT" order="42">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="STRI2INT" order="43">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="nil">nil</arg3>
   </instruction>
   <instruction opcode="STRI2INT" order="44">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="bool">true</arg2>
      <arg3 type="int">8</arg3>
   </instruction>
   <instruction opcode="READ" order="45">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="type">string</arg2>
   </instruction>
   <instruction opcode="READ" order="46">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="type">int</arg2>
   </instruction>
   <instruction opcode="READ" order="47">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="type">bool</arg2>
   </instruction>
   <instruction opcode="WRITE" order="48">
      <arg1 type="var">GF@var</arg1>
   </instruction>
   <instruction opcode="WRITE" order="49">
      <arg1 type="string">aa</arg1>
   </instruction>
   <instruction opcode="WRITE" order="50">
      <arg1 type="int">8</arg1>
   </instruction>
   <instruction opcode="WRITE" order="51">
      <arg1 type="bool">true</arg1>
   </instruction>
   <instruction opcode="WRITE" order="52">
      <arg1 type="nil">nil</arg1>
   </instruction>
   <instruction opcode="CONCAT" order="53">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="CONCAT" order="54">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="CONCAT" order="55">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="nil">nil</arg3>
   </instruction>
   <instruction opcode="CONCAT" order="56">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="bool">true</arg2>
      <arg3 type="int">8</arg3>
   </instruction>
   <instruction opcode="STRLEN" order="57">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
   </instruction>
   <instruction opcode="STRLEN" order="58">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
   </instruction>
   <instruction opcode="STRLEN" order="59">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="nil">nil</arg2>
   </instruction>
   <instruction opcode="STRLEN" order="60">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="int">8</arg2>
   </instruction>
   <instruction opcode="GETCHAR" order="61">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="GETCHAR" order="62">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="GETCHAR" order="63">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="nil">nil</arg3>
   </instruction>
   <instruction opcode="GETCHAR" order="64">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="bool">true</arg2>
      <arg3 type="int">8</arg3>
   </instruction>
   <instruction opcode="SETCHAR" order="65">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="SETCHAR" order="66">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="SETCHAR" order="67">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="nil">nil</arg3>
   </instruction>
   <instruction opcode="SETCHAR" order="68">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="bool">true</arg2>
      <arg3 type="int">8</arg3>
   </instruction>
   <instruction opcode="TYPE" order="69">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="var">GF@var</arg2>
   </instruction>
   <instruction opcode="TYPE" order="70">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="string">aa</arg2>
   </instruction>
   <instruction opcode="TYPE" order="71">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="nil">nil</arg2>
   </instruction>
   <instruction opcode="TYPE" order="72">
      <arg1 type="var">GF@var</arg1>
      <arg2 type="int">8</arg2>
   </instruction>
   <instruction opcode="LABEL" order="73">
      <arg1 type="label">label</arg1>
   </instruction>
   <instruction opcode="JUMP" order="74">
      <arg1 type="label">label</arg1>
   </instruction>
   <instruction opcode="JUMPIFEQ" order="75">
      <arg1 type="label">label</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="JUMPIFEQ" order="76">
      <arg1 type="label">label</arg1>
      <arg2 type="string">aa</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="JUMPIFEQ" order="77">
      <arg1 type="label">label</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="nil">nil</arg3>
   </instruction>
   <instruction opcode="JUMPIFEQ" order="78">
      <arg1 type="label">label</arg1>
      <arg2 type="bool">true</arg2>
      <arg3 type="int">8</arg3>
   </instruction>
   <instruction opcode="JUMPIFNEQ" order="79">
      <arg1 type="label">label</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="JUMPIFNEQ" order="80">
      <arg1 type="label">label</arg1>
      <arg2 type="string">aa</arg2>
      <arg3 type="var">GF@var</arg3>
   </instruction>
   <instruction opcode="JUMPIFNEQ" order="81">
      <arg1 type="label">label</arg1>
      <arg2 type="var">GF@var</arg2>
      <arg3 type="nil">nil</arg3>
   </instruction>
   <instruction opcode="JUMPIFNEQ" order="82">
      <arg1 type="label">label</arg1>
      <arg2 type="bool">true</arg2>
      <arg3 type="int">8</arg3>
   </instruction>
   <instruction opcode="EXIT" order="83">
      <arg1 type="var">GF@var</arg1>
   </instruction>
   <instruction opcode="EXIT" order="84">
      <arg1 type="string">aa</arg1>
   </instruction>
   <instruction opcode="EXIT" order="85">
      <arg1 type="int">8</arg1>
   </instruction>
   <instruction opcode="EXIT" order="86">
      <arg1 type="bool">true</arg1>
   </instruction>
   <instruction opcode="EXIT" order="87">
      <arg1 type="nil">nil</arg1>
   </instruction>
   <instruction opcode="DPRINT" order="88">
      <arg1 type="var">GF@var</arg1>
   </instruction>
   <instruction opcode="DPRINT" order="89">
      <arg1 type="string">aa</arg1>
   </instruction>
   <instruction opcode="DPRINT" order="90">
      <arg1 type="int">8</arg1>
   </instruction>
   <instruction opcode="DPRINT" order="91">
      <arg1 type="bool">true</arg1>
   </instruction>
   <instruction opcode="DPRINT" order="92">
      <arg1 type="nil">nil</arg1>
   </instruction>
   <instruction opcode="BREAK" order="93"/>
</program>
